mlflow/sklearn/__init__.py

```diff
@@ -884,12 +1154,87 @@ def patched_fit(original, self, *args, **kwargs):
                           for autologging (e.g., specify "fit" in order to indicate that
                           `sklearn.linear_model.LogisticRegression.fit()` is being patched)
         """
+        status = _get_autolog_training_status()
         with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
             if t.should_log():
-                return fit_mlflow(original, self, *args, **kwargs)
+                with status.disable_log_post_training_metrics():
+                    result = fit_mlflow(original, self, *args, **kwargs)
+                status.register_model(self, mlflow.active_run().info.run_id)
+                return result
             else:
                 return original(self, *args, **kwargs)
 
+    def patched_predict(original, self, *args, **kwargs):
+        status = _get_autolog_training_status()
+        if status.should_log_post_training_metrics() and status.get_run_id_for_model(self):
+            print(f'DBG: patched_predict # 1')
+            predict_result = original(self, *args, **kwargs)
+            print(f'DBG: patched_predict # 1.5')
+            eval_dataset = args[0] if len(args) >= 1 else kwargs.get('X')
+            eval_dataset_name = status.register_eval_dataset(self, eval_dataset)
+            status.register_prediction_result(
+                self, eval_dataset_name, predict_result)
+            return predict_result
+        else:
+            print(f'DBG: patched_predict # 3')
+            return original(self, *args, **kwargs)
+
+    def patched_metric_api(original, *args, **kwargs):
+        status = _get_autolog_training_status()
+        if status.should_log_post_training_metrics():
+            # one metric api may call another metric api,
+            # to avoid this, call disable_log_post_training_metrics to avoid nested patch
+            with status.disable_log_post_training_metrics():
+                metric = original(*args, **kwargs)
+
+            if status.is_metrics_value_loggable(metric):
+                print('DGB: run patched_metric_api #2')
+
+                metric_name = original.__name__
+                if metric_name.strip() == '<lambda>':
+                    metric_name = 'unknown_metric'
```

Comment:
Is there any change we patch a lambda function?

Reply:
Yes.

Reply:
```suggestion
                metric_name = original.__name__
```

If so, can we remove this if caluse?